src/autoIncrement.ts

```diff
@@ -94,18 +100,28 @@ export function AutoIncrementID(schema: mongoose.Schema<any>, options: AutoIncre
 
     if (!model) {
       logger.info('Creating idtracker model named "%s"', opt.trackerModelName);
-      model = this.db.model(opt.trackerModelName, IDSchema, opt.trackerCollection);
+      model = this.db.model(opt.trackerModelName, IDSchema);
+      model.findOne({ model: opt.model, field: opt.field } as AutoIncrementIDTrackerSpec).lean()
+        .then((counter: AutoIncrementIDTrackerSpec) => {
+          if (!counter) {
+            model.create({
+              model: opt.model,
```

Comment:
i didnt add this *this way* because it would be incorrect, so i used (later in the code) `this.constructor.modelName`

Reply:
we are using multiple models to use auto-increment id. so we have to mention that here.

Reply:
with that i mean it should be got by the actual model later, and not induce some "user error" by inputting the name manually

Reply:
I understand what you are saying. but the plugin `mongoose-auto-increment` where forced us to enter model name and we did that wherever we need auto-increment id with different name. I mean, collection name and the model name will be different in that `identitycounters` collection.